tests/test_factory.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/python
```

Comment:
This should be unnecessary. If you put them in the right directory, run_tests.py should discover and run them for you.